tasks/libs/pipeline_data.py

```diff
@@ -130,3 +130,19 @@ def truncate_job_name(job_name, max_char_per_job=48):
     # We also want to avoid it being too long
     truncated_job_name = truncated_job_name[:max_char_per_job]
     return truncated_job_name
+
+
+# those jobs are `allow_failure: true` but still needs to be included
+# in fail reports
+jobs_allowed_to_fails_but_need_report = [re.compile(r'kitchen_test_security_agent.*')]
+
+
+def should_include_failed_job(job_name, allow_failure):
+    if not allow_failure:
+        return True
+
+    for job_to_include in jobs_allowed_to_fails_but_need_report:
+        if job_to_include.fullmatch(job_name):
+            return True
+
+    return False
```

Comment:
```suggestion
    return any(pattern.fullmatch(job_name) for pattern in jobs_allowed_to_fails_but_need_report)
```
this is a more pythonic way for line 144-148